src/core/ext/transport/chttp2/server/chttp2_server.cc

```diff
@@ -86,6 +86,20 @@ class Chttp2ServerListener : public Server::ListenerInterface {
 
   void Orphan() override;
 
+  // The interface of RefCounted<> has been manually implemented here to take a ref on tcp_server_ instead. Note that, the handshaker needs tcp_server_ to exist for the lifetime of the handshake since it's needed by acceptor. Sharing refs between the listener and tcp_server_ is just an optimization to avoid taking additional refs on the listener, since TcpServerShutdownComplete already holds a ref to the listener.
+  void IncrementRefCount() {
```

Comment:
Is the suggestion to log the location of the refs and unrefs too protected by a new TraceFlag?

Reply:
I'm not necessarily suggesting that we add a new TraceFlag.  I was just thinking that if we're going to claim that this object supports the interface expected by `RefCountedPtr<>`, then we should support all of that interface, even if parts of it aren't currently being used, since they may be used in the future.

Reply:
Done